processor/src/main/java/com/github/nmorel/gwtjackson/rest/processor/RestService.java

```diff
@@ -78,18 +81,23 @@ public RestService( Options options, Element element ) {
 
         Path path = typeElement.getAnnotation( Path.class );
         String baseRestUrl = path.value();
+        
+        Consumes baseConsumes = typeElement.getAnnotation( Consumes.class );
+        Produces baseProduces = typeElement.getAnnotation( Produces.class );
 
         for ( ExecutableElement method : ElementFilter.methodsIn( typeElement.getEnclosedElements() ) ) {
-            parseMethod( baseRestUrl, method );
+            parseMethod( baseRestUrl, baseConsumes, baseProduces, method );
         }
     }
 
-    private void parseMethod( String baseRestUrl, ExecutableElement method ) {
-        AnnotationMirror httpMethodAnnotation = isRestMethod( method );
+    private void parseMethod( String baseRestUrl, Consumes baseConsumes, Produces baseProduces, ExecutableElement method ) {
+        AnnotationMirror httpMethodAnnotation = isRestMethod( method, baseConsumes, baseProduces );
         if ( null == httpMethodAnnotation ) {
             // not a rest method
             return;
         }
+        String consumes = getFirstJsonType( getConsumes( baseConsumes, method ) );
```

Comment:
do the getFirstJsonType inside getConsumes and getProduces methods ? You call it each time after the method.